Build pipeline and .NET 3.5 support

CodeHex opened the pull request:
This is my attempt at adding automated build scripts. I needed to get multiple frameworks supported in the Nuget packages, so I've also added .NET 3.5 support (so should solve #25). It works by having a separate project for .NET 3.5 so it can target that framework, but just links the original files.

The HowToBuild.md in the build folder is a good starting place. I've linked my forked repo to AppVeyor so you can see the output (https://ci.appveyor.com/project/CodeHex/bugsnag-dotnet). It also builds a Nuget package for the Mono version of Bugsnag which I managed to quickly test in Xamarin Studio.

Let me know what you think, not really sure if its along the same lines of your existing build process for the other apps.

Need to test it a bit more, but thought I'd put the pull request in so you can look at it.

snmaynard commented:
This looks good - does this mean that we will always have two nuget packages (mono and .net) with the .net package bundling support for as many framework versions as we want?

This is slightly different to how we build for other platforms, but in a good way. Usually we just manually build on our dev machines, but we dont run windows so having it automatically producing build artifacts will be incredibly useful for us.

CodeHex commented:
That's right, but both the normal version and the Mono version would support different .NET frameworks. That way, e.g. if you created a library for .NET 3.5 and wanted it to run on Mono as well, you could use the BugsnagMono nuget package, which would use the Mono/.NET 3.5 compatible version for that library. That's the idea anyway. The build script would handle testing each variation automatically.

The AppVeyor way of building should hopefully let you build the artifacts without needing a Windows environment.

CodeHex commented:
Looks like the IconUrls in the nuget packages need to point to 32x32 icons or they don't show. I've added a local .png logo file and updated the links to file in github if this gets merged into master.

If you want them to point to something else, just need to delete the logo .png and update the nuspec links.
